app/services/katello/pulp3/migration_plan.rb

```diff
@@ -105,7 +105,15 @@ def target_link_map
       end
 
       def name_for_content_view(content_view, root_repo)
-        "#{content_view.label}-#{root_repo.label}"
+        name = "#{content_view.label}-#{root_repo.label}"
+
+        if Katello::RootRepository.where(:label => root_repo.label).group(:label).count(:label)[root_repo.label] > 1
+          repo_query = Katello::Repository.joins(:root, :content_view_version => :content_view).
+              where("#{::Katello::RootRepository.table_name}.label" => root_repo.label, "#{::Katello::ContentView.table_name}.label" => content_view.label)
```

Comment:
Wouldn't it be better to select directly for the content_view.id instead as the content_view is a parameter to this function?

Reply:
i'll write up an explanation about how the data looks and why (i think) this is a good solution